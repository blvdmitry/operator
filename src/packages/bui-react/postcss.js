// Re-export of the config provided by bui-postcss package
const config = require("@bookingcom/bui-postcss");

module.exports = config.default;
