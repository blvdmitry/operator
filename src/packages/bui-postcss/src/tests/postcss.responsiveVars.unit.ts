/* eslint-disable import/no-extraneous-dependencies */
import postcss from "postcss";

const responsiveVarsPlugin = require("../postcss.responsiveVars");

test("postcss.responsiveVars", async () => {
  const input = `
    @bui-responsive-vars .bui_mixin_width, mixin, width {
      @bui-unit;
      @bui-literal;
    };

    @bui-responsive-vars .bui_mixin_position, mixin, position {
      @bui-literal;
      @bui-important;
    };

    @bui-responsive-vars .bui_mixin_position, mixin, position;;
  `;
  const output = await postcss([responsiveVarsPlugin()]).process(input, {
    from: "__placeholder-name.css",
  });

  expect(output.css).toMatchSnapshot();
});
