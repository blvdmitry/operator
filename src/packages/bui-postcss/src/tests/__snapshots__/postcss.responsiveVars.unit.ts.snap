// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcss.responsiveVars 1`] = `
".bui_mixin_width{

    width: var(--bui_mixin_width)
}

.bui_mixin_width[style*="--bui_mixin_width--s"]{

    --bui_mixin_width: var(--bui_mixin_width--s)
}

.bui_mixin_width[style*="--bui_mixin_spaced_width--s"]{

    --bui_mixin_width: calc(var(--bui_mixin_spaced_width--s) * var(--bui_spacing_1x))
};

    @media (--bui_medium_viewport){

    .bui_mixin_width[style*="--bui_mixin_width--m"]{

        --bui_mixin_width: var(--bui_mixin_width--m)
    }

    .bui_mixin_width[style*="--bui_mixin_spaced_width--m"]{

        --bui_mixin_width: calc(var(--bui_mixin_spaced_width--m) * var(--bui_spacing_1x))
    }
};

    @media (--bui_large_viewport){

    .bui_mixin_width[style*="--bui_mixin_width--l"]{

        --bui_mixin_width: var(--bui_mixin_width--l)
    }

    .bui_mixin_width[style*="--bui_mixin_spaced_width--l"]{

        --bui_mixin_width: calc(var(--bui_mixin_spaced_width--l) * var(--bui_spacing_1x))
    }
};

    @media (--bui_xlarge_viewport){

    .bui_mixin_width[style*="--bui_mixin_width--xl"]{

        --bui_mixin_width: var(--bui_mixin_width--xl)
    }

    .bui_mixin_width[style*="--bui_mixin_spaced_width--xl"]{

        --bui_mixin_width: calc(var(--bui_mixin_spaced_width--xl) * var(--bui_spacing_1x))
    }
};

    .bui_mixin_position{

    position: var(--bui_mixin_position) !important
};

    .bui_mixin_position[style*="--bui_mixin_position--s"]{

    --bui_mixin_position: var(--bui_mixin_position--s)
};

    @media (--bui_medium_viewport){

    .bui_mixin_position[style*="--bui_mixin_position--m"]{

        --bui_mixin_position: var(--bui_mixin_position--m)
    }
};

    @media (--bui_large_viewport){

    .bui_mixin_position[style*="--bui_mixin_position--l"]{

        --bui_mixin_position: var(--bui_mixin_position--l)
    }
};

    @media (--bui_xlarge_viewport){

    .bui_mixin_position[style*="--bui_mixin_position--xl"]{

        --bui_mixin_position: var(--bui_mixin_position--xl)
    }
};

    @bui-responsive-vars .bui_mixin_position, mixin, position;;
  "
`;
